### PR TITLE
add lc 1329

### PR DESCRIPTION
**Pull request by snowan**

### CODE REVIEW

**Comment by snowan** [comment]
@sunboman review

File: 2021/python/1329.sort-matrix-diagonal.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+Solution: intuitive solution, list of diagonal values, and sort, replace back into matrix.
+the same diagonal m[r][c] has same value of r - c.
+
+"""
+def diagonalSort(mat: List[List[int]]) -> List[List[int]]:
+    if not mat or len(mat) == 0 or len(mat[0]) == 0:
+        return mat
+    m, n = len(mat), len(mat[0])
+    # dictionary to store same r-c into list
+    diag_map = {}
```
Comment (sunboman) [inline 2021/python/1329.sort-matrix-diagonal.py]: `defaultlist(list)`

File: 2021/python/1329.sort-matrix-diagonal.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    for r in range(m):
+        for c in range(n):
+            diag_map[r - c].append(mat[r][c])
```
Comment (sunboman) [inline 2021/python/1329.sort-matrix-diagonal.py:32]: is this working? what if `r-c` not existed in the map yet?

Comment (snowan) [reply 2021/python/1329.sort-matrix-diagonal.py:32]: --yes, this is working.--
my bad, map init should be `defaultdict(list)` otherwise it is not working.

Comment (sunboman) [reply 2021/python/1329.sort-matrix-diagonal.py:32]: you are wrong!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

**Review by sunboman — APPROVED** [review]

**Pull request merged**